### PR TITLE
tests,utils: Remove GenerateVMJson helper

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Remove GenerateVMJson e2e util function, inline its code where used.

Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/vm_test.go [hunk 1]
```diff
@@ -991,8 +992,11 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine(), opts...)
 			vm.Namespace = testsuite.GetTestNamespace(vm)
 
-			vmJson, err := tests.GenerateVMJson(vm, workDir)
-			Expect(err).ToNot(HaveOccurred(), "Cannot generate VMs manifest")
+			data, err := json.Marshal(vm)
```
Comment (dankenigsberg) [inline tests/vm_test.go:995]: In my opinion, we should not be in the business of testing `kubectl`. Can you you please delete the tests that use this function instead, e.g in a follow-up pr?

Comment (0xFelix) [reply tests/vm_test.go:995]: Somewhere there are tests that verify the columns printed by `kubectl` look like we are expecting them too. IMO that is a valid test case. But these tests should probably dropped.

See https://github.com/kubevirt/kubevirt/issues/12709

**Review by 0xFelix — COMMENTED** [review]
/approve

File: tests/vm_test.go (hunk 1 above)
Comment (dankenigsberg) [reply tests/vm_test.go:995]: (unrelated to this PR)

> Somewhere there are tests that verify the columns printed by kubectl look like we are expecting them too. IMO that is a valid test case

We may want to have an integration test with `kubectl`, but I don't think it makes sense to start a VM (with all the time, race condition and wasted resources) just for that.

> But these tests should probably dropped.

Yes. `GenerateVMJson` was a code smell.

**Comment by ormergi** [comment]
[Changes](https://github.com/kubevirt/kubevirt/compare/8a2252aa37f3b4d63486f1f19087e716894a8b1c..3757f07f499561d8a47567ed22bac2e2a87985ae): Rebased

**Review by arnongilboa — COMMENTED** [review]
@ormergi can you please re-explain why  we better remove this helper and duplicate its code 3 times?

**Comment by ormergi** [comment]
> @ormergi can you please re-explain why we better remove this helper and duplicate its code 3 times?

@arnongilboa  The intention is to removing this helper following the effort for cleaning up the junk-plie test/utils.go.
In this case the logic is very simple, I dont think see advantage in move the code to some package.
In addition following these PRs https://github.com/kubevirt/kubevirt/pull/13026 https://github.com/kubevirt/kubevirt/pull/13029 the usages of this function is reduced to 1.

**Pull request merged**